core/model/modx/processors/search/search.class.php

```diff
@@ -18,37 +18,651 @@ class modSearchProcessor extends modProcessor
     public function process()
     {
         $this->query = $this->getProperty('query');
+        
         if (!empty($this->query)) {
             if (strpos($this->query, ':') === 0) {
-                // upcoming "launch actions"
-                //$this->searchActions();
+                // Advanced Search/Actions
+                
+                $query = explode(':',$this->query);
+                $this->advancedSearch($query);
+                
             } else {
-                // Search elements & resources
-                $this->searchResources();
-                if ($this->modx->hasPermission('view_chunk')) {
-                    $this->searchChunks();
+                // Default search only in resources
+                $query = explode(':',$this->query);
+                
+                $query = ':r:*:'.$query[0] . ':'.$query[1];
```

Comment:
This part can only search through resources but I agree that this look very ugly. I'll wrap this part up in the next minutes.